stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/avatar/AvatarView.kt

```diff
@@ -0,0 +1,131 @@
+package io.getstream.chat.android.ui.avatar
+
+import android.content.Context
+import android.graphics.Canvas
+import android.graphics.Color
+import android.graphics.Paint
+import android.graphics.Path
+import android.graphics.RectF
+import android.util.AttributeSet
+import androidx.appcompat.widget.AppCompatImageView
+import io.getstream.chat.android.client.models.Channel
+import io.getstream.chat.android.client.models.User
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.launch
+
+public class AvatarView @JvmOverloads constructor(
+    context: Context,
+    attrs: AttributeSet? = null,
+    defStyleAttr: Int = 0
+) : AppCompatImageView(context, attrs, defStyleAttr) {
+    private val bitmapFactory = AvatarBitmapFactory(context)
+    private val clipPath = Path()
+    private val clipRect = RectF()
+    private val borderPaint = Paint(Paint.ANTI_ALIAS_FLAG).apply {
+        style = Paint.Style.STROKE
+    }
+    private val onlineIndicatorOutlinePaint = Paint().apply {
+        style = Paint.Style.FILL
+        color = Color.WHITE
+    }
+    private val onlineIndicatorPaint = Paint().apply {
+        style = Paint.Style.FILL
+        color = Color.GREEN
+    }
+
+    private lateinit var avatarStyle: AvatarStyle
+    private var onlineIndicatorVisible: Boolean = false
+
+    init {
+        setStyle(AvatarStyle(context, attrs))
+    }
+
+    public fun setChannelData(channel: Channel, user: List<User>) {
+        configureImageDrawable {
+            AvatarDrawable(bitmapFactory.createChannelBitmaps(channel, user, avatarStyle))
+        }
+    }
+
+    public fun setUserData(user: User) {
+        configureImageDrawable {
+            AvatarDrawable(listOfNotNull(bitmapFactory.createUserBitmap(user, avatarStyle)))
+        }
+    }
+
+    public fun toggleOnlineIndicatorVisibility(visible: Boolean) {
+        this.onlineIndicatorVisible = visible
+        invalidate()
+    }
+
+    public fun setStyle(avatarStyle: AvatarStyle) {
```

Comment:
Does this have to be `public`, do we expect clients to create instances of `AvatarStyle` programatically? If not, that entire class could be `internal` too.

Reply:
Still looks to me like `AvatarStyle` should be `internal`.

Reply:
Made AvatarStyle internal